### ISSUE TITLE
Phase 3, overlap loss

### ISSUE THREAD

**Issue by haoala** [czq142857/BSP-NET-pytorch#3]
Thank you for your highly interesting work and for releasing the code. I have a few questions.

I see in the code that there is a training loss not described in the paper, the one where you encourage the values in **_T_** to move towards 0 or 1. How does this loss perform? If using phase 3, should it be after phase 0 training or should we start fro, scratch with phase 3?

Also, I saw that the implementation of the overlap loss in the code seems different from what is described in the paper. What is the reason for this difference?

Finally, are the pretrained network weights you provided trained with or without the overlap loss?

**Comment by czq142857** [comment]
1. There are a lot of things we wanted to try but did not try due to the paper submission deadline. Phase 3 was one of them. I tried phase 3 afterward, and it usually yielded better results. If you want to try phase 3, you need to place it after phase 0. Here are the results (single view reconstruction) in LFD (which is more consistent with my own visual evaluation).
[Original implementation]
phase 0 -> phase 1 : 2939.1472
phase 0 -> phase 2 : 2939.5001
phase 0 -> phase 3 : 2911.6589
[Pytorch implementation]
phase 0 -> phase 1 : 2945.7945

2. This is my mistake and I really need to thank you for this. The overlap loss in the paper is wrongly written. I did not notice it during several proof-reading (probably because this loss term is not that important for 3D). Anyway, I corrected the equation in ArXiv and now it reads:
![image](https://user-images.githubusercontent.com/24209790/91780528-a6966d80-ebac-11ea-93a6-d3dbf99e684d.png)

There is still a slight difference between the equation and the implementation. You can use the following loss if you want to follow the paper. I am sure the results will be very similar.
```python
self.loss = self.loss_sp - tf.reduce_mean(self.G2*self.bmask)
```

The implementation in my code also incorporates ground truth values, which may prevent over-punishing wrongly-predicted points (i.e., those points that are predicted as inside but are actually outside), because those points are already punished by the reconstruction loss.
```python
self.loss = self.loss_sp - tf.reduce_mean(self.G2*self.point_value*self.bmask)
```

3. The pretrained network weights is without the overlap loss (phase 0 -> phase 1).

Best,
Zhiqin
